src/pushover_complete/pushover_api.py

```diff
@@ -173,7 +183,7 @@ def send_message(self, user, message, device=None, title=None, url=None, url_tit
         :returns: Response body interpreted as JSON
         :rtype: dict
         """
-        return self._send_message(user, message, device, title, url, url_title, priority, retry, expire, callback_url,
+        return self._send_message(user, message, device, title, url, url_title, image_path, priority, retry, expire, callback_url,
```

Comment:
I'm gonna request a semi-major change here: can the argument be changed to `image` and we allow either `str or pathlib.Path` **or** a `BytesIO` object? That would allow an image to be created in memory by the application and sent without ever writing it to disk. It would also allow the caller to manage the lifetime of the file handle.

This is just a change to the signature and docs of `send_message` but would require a bit more logic in `_send_message` to decide whether to open the file or just pass it along.

Shouldn't change `_generic_post` at all, I think.

Reply:
what's the best way to do this and keep the code clean? Have two versions of the function that handle the file argument differently then the handler picks which strategy based on the type of the input?

Reply:
Ah, yes. I can do that, but I probably won't get to it until this evening. I've pushed the other changes for now and I'll get to this as soon as I can.

Reply:
I found the time to put it together right now. That said, I haven't tested it beyond passing in a string still works.